### PR TITLE
Generalizing script for more tables (namely 'shorturl' and 'geoadmin-file-storage')

### PR DESCRIPTION
**Pull request by procrastinatio**
Generalizing both script 'dump.py' and 'restore.py' to be used with many dynamoDB tables.

### CODE REVIEW

**Comment by procrastinatio** [comment]
@loicgasser Could you have a look... It is a bit scary.

File: restore.py [hunk 1]
```diff
@@ -57,28 +63,44 @@ def write_data(file_name):
     finally:
         file_data.close()
 
+def usage():
+    print "Restore a dumped dynamoBD to the given 'table' in 'eu-west-1'"
+    print ""
+    print "Usage: restore.py [-h --help] [-t --table=table_name] [-c --create] [-f --filter=20140805] table_name"
```
Comment (loicgasser) [inline restore.py]: remove table_name at the end

Comment (loicgasser) [inline restore.py]: The

File: dump.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#!/usr/bin/env python
```
Comment (loicgasser) [inline dump.py]: no need

File: dump.py [hunk 3]
```diff
@@ -170,3 +176,28 @@ def manage_retention(dumpDir):
...
     logger.info('DUMP_STATUS=0')
+
+def usage():
+    print "Dump a dynamoDB table in 'eu-west-1' to local filesystem"
+    print ""
+    print "Usage: dump.py [-t table_name]"
+
+def main(argv):
+
+    TABLE_NAME = 'shorturl'
```
Comment (loicgasser) [inline dump.py]: So this is the default value so that we don't need to adapt the cron job?

File: dump.py [hunk 4]
```diff
@@ -121,25 +130,22 @@ def manage_retention(dumpDir):
...
-if __name__ == '__main__':
+def dump(table_name):
     t0 = time.time()
-    DUMP_DIR = '/var/backups/dynamodb/'
-    PREFIX_NAME = 'data_'
-    TABLE_NAME = 'shorturl'
-    JSON_INDENT = 2
-    FOLDER_NAME = datetime.datetime.fromtimestamp(t0).strftime('%Y%m%d')
+    folder_name = datetime.datetime.fromtimestamp(t0).strftime('%Y%m%d')
+    dump_dir = os.path.join(BASE_DIR, table_name, folder_name)
```
Comment (loicgasser) [inline dump.py]: ok so you want to create a subfolder in /var/backups/dynamodb/ based on the table name?

Comment (procrastinatio) [inline dump.py]: Yes, we'll have ` /var/backups/dynamodb/shortul`and ` /var/backups/dynamodb/filestorage`

File: dump.py (hunk 3 above)
Comment (procrastinatio) [inline dump.py]: Yes, for both script the default value goes to `shorturl`. Behaviour should remained unchanged...

File: dump.py (hunk 4 above)
Comment (loicgasser) [inline dump.py]: ok good

File: restore.py [hunk 5]
```diff
@@ -135,8 +157,22 @@ def write_data(file_name):
             )
             ## Wait 30 secs for the table to create
             time.sleep(30)
+        elif CREATE_TABLE and TABLE_NAME == 'geoadmin-filestorage':
```
Comment (loicgasser) [inline restore.py]: not sure about the table name here

**Comment by loicgasser** [comment]
please adapt README and comments in the code as well.

File: dump.py (hunk 4 above)
Comment (loicgasser) [inline dump.py]: so you actually need to create `shortul` and `geoadmin-file-storage` folders.

Comment (loicgasser) [inline dump.py]: before creating 20150503

Comment (procrastinatio) [inline dump.py]: `os.makedirs`(similar to `mkdir -p`) do the trick

**Comment by procrastinatio** [comment]
OK, I have tested `dump.py`and `restore.py`against `geoadmin-file-storage` and it works as expected. The trick is that we can only update the `throuput`four time a day.

**Comment by procrastinatio** [comment]
@loicgasser @gjn no volunteer to merge ?

**Comment by loicgasser** [comment]
@procrastinatio update in ´/home/dynamodb´ must be performed manually otherwise your changes won't be taken into accoutnt. (using dynamodb user)

**Comment by procrastinatio** [comment]
OK @loicgasser thank you.

**Pull request merged**